include/dts-functions.sh

```diff
@@ -1387,3 +1387,70 @@ show_menu() {
     echo -ne "${RED}${SSH_OPT_UP}${NORMAL} to launch SSH server  ${NORMAL}"
   fi
 }
+
+check_if_fused() {
+file_path="/sys/class/mei/mei0/fw_status"
+
+if [[ ! -f $file_path ]]; then
+    echo "File not found: $file_path"
+    return 2
+fi
+
+hfsts6_value=""
+line_number=1
+while IFS= read -r line; do
+    if [[ $line_number -eq 6 ]]; then
+        hfsts6_value=$line
+        break
+    fi
+    ((line_number++))
+done < "$file_path"
+
+if [[ -z $hfsts6_value ]]; then
+    echo "Failed to read HFSTS6 value"
+    exit 1
+fi
+
+hfsts6_binary=$(echo "ibase=16; obase=2; $hfsts6_value" | bc)
+
+binary_length=${#hfsts6_binary}
+
+# Add leading zeros
+if [ $binary_length -lt 32 ]; then
+  padding=$((32 - binary_length))
+  zeros=$(printf "%${padding}s" | tr ' ' "0")
+  hfsts6_binary=$zeros$hfsts6_binary
+fi
+
+bit_30_value=${hfsts6_binary:1:1}
+
+if [ $bit_30_value == 0 ]; then
+  return 1
+else
+  return 0
+fi
+}
+
+check_if_boot_guard_enabled() {
+  # Get MSR in binary
+  msr=$(rdmsr 0x13a -0 | xxd -b | awk '{for(i=2; i<=NF; i++) if($i ~ /^[01]+$/) printf "%s", $i}' | tr -d '\n')
```

Comment:
I gave it a quick test on my laptop. The rdmsr gives me:

```
sudo /usr/sbin/rdmsr 0x13a -0
000000130000007d
```

while this function gives:

```
+ check_if_boot_guard_enabled
++ /usr/sbin/rdmsr 0x13a -0
++ xxd -b
++ awk '{for(i=2; i<=NF; i++) if($i ~ /^[01]+$/) printf "%s", $i}'
++ tr -d '\n'
+ msr=0011000000110000001100000011000000110000001100000000000011000100110011001100000011000000110000001100000011000000110000001101110110010000001010
+ facb_fpf=0
+ verified_boot=1
+ '[' 0 == 1 ']'
+ return 1
```

The binary value of the MSR does not match the hex value from the rdmsr output. Thus the facb and verified boot values are incorrect (both should be 1 in my case.

On a platform with BtG disabled:

```
rdmsr 0x13a -0
0000000000000000
```

And the script:

```
+ check_if_boot_guard_enabled
++ rdmsr 0x13a -0
++ xxd -b
++ awk '{for(i=2; i<=NF; i++) if($i ~ /^[01]+$/) printf "%s", $i}'
++ tr -d '\n'
+ msr=0011000000110000001100000011000000110000001100000000000011000000110000001100000011000000110000001100000000000011000000110000001100000011000000001010
+ facb_fpf=0
+ verified_boot=1
+ '[' 0 == 1 ']'
+ return 1
```

Reply:
Also why using a different converstion than here?
`hfsts6_binary=$(echo "ibase=16; obase=2; $hfsts6_value" | bc)`

Reply:
Should be ok now 8bb0b85a8528df8e3bbbd7815ff9772ea6c6c18d

Reply:
Still doesn't work:

```
+ check_if_boot_guard_enabled
+ rdmsr 0x13a -0
000000130000007d
++ rdmsr 0x13a -0
+ msr_hex=000000130000007d
++ echo 'ibase=16; obase=2; 000000130000007d'
++ bc
(standard_in) 1: syntax error
+ msr_binary=
+ binary_length=0
+ '[' 0 -lt 64 ']'
+ padding=64
++ tr ' ' 0
++ printf %64s
+ zeros=0000000000000000000000000000000000000000000000000000000000000000
+ msr_binary=0000000000000000000000000000000000000000000000000000000000000000
+ btg_capability=0
+ facb_fpf=0
+ verified_boot=0

```

Reply:
It was a matter of setting the hex string to uppercase d407c14bfb8efaccb8475b979738f1e46d6bffa6

Reply:
Now it seems to be working well.